src/test/java/org/opentripplanner/openstreetmap/OSMOpeningHoursParserTest.java

```diff
@@ -0,0 +1,178 @@
+package org.opentripplanner.openstreetmap;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import ch.poole.openinghoursparser.OpeningHoursParseException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.opentripplanner.model.calendar.openinghours.OpeningHoursCalendarService;
+import org.opentripplanner.routing.trippattern.Deduplicator;
+import org.opentripplanner.test.support.VariableSource;
+
+public class OSMOpeningHoursParserTest {
+
+  static OpeningHoursCalendarService openingHoursCalendarService = new OpeningHoursCalendarService(
+    new Deduplicator(),
+    LocalDate.of(2022, Month.SEPTEMBER, 1),
+    LocalDate.of(2023, Month.DECEMBER, 31)
+  );
+
+  static OSMOpeningHoursParser osmOpeningHoursParser = new OSMOpeningHoursParser(
+    openingHoursCalendarService,
+    ZoneId.of("Europe/Paris")
+  );
+
+  static Stream<Arguments> osmOpeningHoursTestCases = Stream.of(
```

Comment:
Can you add a test case for 24/7?

Reply:
There are a couple (hard to spot because of how the lines are split) that include the 24/7 and then there are there is one that has rule that is just `open` which in practice is the same thing. I'm not sure what we should do about opening hour rules that are just a random comment modifier like "by appointment" (there is a test for it but currently it's interpreted as being closed).

Reply:
I think those "by appointment" ones are totally out of the scope of OTP. I'd be fine with just ignoring it. You can implement whatever you find fits best.

One cannot expect to convert very imprecise input into precise output.

Reply:
If there isn't any opening hours definition, we just guess it's open 24 hours because we don't have any other information. In some ways an opening hours definition that is just a comment is a similar case where we don't really know but maybe it's a bit more likely that the comment sets some limits for the opening hours.